### PR TITLE
AB-418 removed redirect from map on mouse click

### PR DESCRIPTION
**Pull request by FeelsGOOD**
We removed redirect from map image with keyboard, so we supposed to delete redirect with mouse as well, because we have above button with the same.

### CODE REVIEW

**Comment by kelvingraddick** [comment]
@FeelsGOOD Hey! I know this was the suggested solution from the audit report but I don't think it's the right solution. I think we need to rethink this with Hadas/Talyah/Brett/Elad/etc. Removing the mouse click event to open Google Maps will regress functionality for any brand currently using the map. Brands can use the map outside of a carousel and without a button under it; in that case there is no ability to navigate. 
I think the solution should be to make the map tabable and/or a button itself, since by default clicking it always performs an action.

File: js/ElementRendererProvider.js [hunk 1]
```diff
@@ -357,9 +358,17 @@ export default class ElementRendererProvider {
...
-      if (config.accessibility && config.accessibility.web) {
+  
+      if (accessibilityWeb) {
         Utils.appendAttributesFromObject(divEl, config.accessibility.web);
+      } else if ((accessibilityWeb && !config.accessibility.web.tabindex) || !config.accessibility) {
+        divEl.setAttribute("tabindex", "0");
+        divEl.addEventListener("keyup", function(event) {
+          event.preventDefault();
+          if (event.keyCode === 13 || event.keyCode === 32) {
+            window.open(`https://www.google.com/maps/search/?api=1&query=${config.la},${config.lo}`, '_blank');
```
Comment (WaveLinkLLC) [inline js/ElementRendererProvider.js:369]: @FeelsGOOD I think here we should include the action click logic from lines below (i.e. `if (config.click && config.click.actions) { this.wrapAction(config.click); ...`

**Comment by kelvingraddick** [comment]
@FeelsGOOD I think here we should include the action click logic from lines below (i.e. if (config.click && config.click.actions) { this.wrapAction(config.click); ...

File: js/ElementRendererProvider.js [hunk 2]
```diff
@@ -357,9 +358,17 @@ export default class ElementRendererProvider {
       if (config.style) {
         divEl.style.cssText = Utils.styleToCss(config.style);
       }
-
-      if (config.accessibility && config.accessibility.web) {
+  
+      if (accessibilityWeb) {
         Utils.appendAttributesFromObject(divEl, config.accessibility.web);
+      } else if ((accessibilityWeb && !config.accessibility.web.tabindex) || !config.accessibility) {
```
Comment (kelvingraddick) [inline js/ElementRendererProvider.js:364]: @FeelsGOOD can you explain your approach with the if statement here?

File: js/ElementRendererProvider.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pprabesh) [reply js/ElementRendererProvider.js:369]: @FeelsGOOD @kelvingraddick Correct me if i'm wrong, but. I think this was just added for keyboard action to follow tab navigation and not click. So this should be fine. The click action already is already defined elsewhere.

File: js/ElementRendererProvider.js (hunk 2 above)
Comment (pprabesh) [reply js/ElementRendererProvider.js:364]: @kelvingraddick  Based on my understanding, i think we are adding default tabindex="0" for maps if there is no external accessibility json in the structured content. Can you confirm @FeelsGOOD

File: js/ElementRendererProvider.js [hunk 3]
```diff
@@ -358,9 +359,18 @@ export default class ElementRendererProvider {
         divEl.style.cssText = Utils.styleToCss(config.style);
       }
 
-      if (config.accessibility && config.accessibility.web) {
+      if (accessibilityWeb) {
         Utils.appendAttributesFromObject(divEl, config.accessibility.web);
       }
+      if ((accessibilityWeb && !accessibilityWeb.tabindex) || !config.accessibility) {
+        divEl.setAttribute('tabindex', '0');
+        divEl.onkeydown = (event) => {
```
Comment (powerbot15) [inline js/ElementRendererProvider.js:367]: @FeelsGOOD I think keydown listener should be moved out of the `if` statement. It should be added anyway regardless from we have something in `accessibility.web` or not.

File: js/ElementRendererProvider.js [hunk 4]
```diff
@@ -358,9 +359,18 @@ export default class ElementRendererProvider {
         divEl.style.cssText = Utils.styleToCss(config.style);
       }
 
-      if (config.accessibility && config.accessibility.web) {
+      if (accessibilityWeb) {
         Utils.appendAttributesFromObject(divEl, config.accessibility.web);
       }
+      if ((accessibilityWeb && !accessibilityWeb.tabindex) || !config.accessibility) {
```
Comment (powerbot15) [inline js/ElementRendererProvider.js:365]: @FeelsGOOD this statement can be simplified. It should be enough to check only 
```javascript
if(!accessibilityWeb || !accessibilityWeb.tabindex)
```

**Review by powerbot15 — CHANGES_REQUESTED** [review]

**Review by powerbot15 — APPROVED** [review]

**Pull request merged**